pybind_interface/pybind_main.cpp

```diff
@@ -542,6 +735,75 @@ std::vector<unsigned> qsim_sample(const py::dict &options) {
   return result_bits;
 }
 
+std::vector<unsigned> qtrajectory_sample(const py::dict &options) {
+  NoisyCircuit<Cirq::GateCirq<float>> ncircuit;
+  unsigned num_qubits;
+  try {
+    ncircuit = getNoisyCircuit(options);
+    num_qubits = qsim::count_qubits(ncircuit);
+  } catch (const std::invalid_argument &exp) {
+    IO::errorf(exp.what());
+    return {};
+  }
+
+  using Simulator = qsim::Simulator<For>;
+  using StateSpace = Simulator::StateSpace;
+  using State = StateSpace::State;
+  using Runner = qsim::QuantumTrajectorySimulator<IO, Cirq::GateCirq<float>,
+                                                  MultiQubitGateFuser,
+                                                  Simulator>;
+
+  Runner::Parameter param;
+  uint64_t seed;
+  try {
+    param.num_threads = parseOptions<unsigned>(options, "t\0");
+    param.max_fused_size = parseOptions<unsigned>(options, "f\0");
+    param.verbosity = parseOptions<unsigned>(options, "v\0");
+    seed = parseOptions<unsigned>(options, "s\0");
+    param.collect_mea_stat = true;
+  } catch (const std::invalid_argument &exp) {
+    IO::errorf(exp.what());
+    return {};
+  }
+
+  std::vector<std::vector<unsigned>> results;
```

Comment:
Would it be worth reserving sizes here ?

Reply:
I don't think we have cheap access to # of measurements here. Unles we just want to reserve an arbitrary, large amount of space, it's probably not worth it.